types/src/account/associated_keys.rs

```diff
@@ -28,16 +26,14 @@ impl AssociatedKeys {
 
     /// Adds new AssociatedKey to the set.
     /// Returns true if added successfully, false otherwise.
-    #[allow(clippy::map_entry)]
     pub fn add_key(&mut self, key: AccountHash, weight: Weight) -> Result<(), AddKeyFailure> {
```

Comment:
Seems like it might be safer to take the `max_associated_keys` as a parameter here, so there's no chance of forgetting to do the check at the callsites.  I realise for now there's only one callsite and the check is done correctly, so this suggestion is maybe overkill.

Reply:
I see your point, but I'd prefer it to remain validated from EngineState's side - I think the limit is valid for every instance of Account (and AssociatedKeys) created in the system rather than a method